core/base/dispatch_helper.hpp

```diff
@@ -13,20 +13,55 @@
 
 
 namespace gko {
+namespace detail {
```

Comment:
This is already an internal header, does it need to be in the `detail` namespace ?

Reply:
We also use that namespace in our private files from time to time. Here I use it because the `run_impl` functions should not be used directly and are just an implementation detail.